### DISCUSSION TITLE
stringly-typed AttributeAccessor

### DISCUSSION THREAD

**Discussion by gavinking** [hibernate/hibernate-orm#3780] [Design Proposals]
`AttributeAccessor` has a stringly-typed `value` member. This should be a `Class`, of course, but how should we make that change? 

- Add a new member and deprecate the existing one? 
- Add a whole new annotation and deprecate the existing one? 
- Or just change it and break people?
- Something else?

**Comment by sebersole** [comment]
Personally I'd vote for new+deprecate

**Reply by gavinking** [reply]
> Personally I'd vote for new+deprecate

Annotation or member?

**Reply by sebersole** [reply]
Member

**Reply by gavinking** [reply]
Fine by me. `@AttributeAccessor(type=TheClass)` ?

**Reply by sebersole** [reply]
`@AttributeAccessor(strategy=TheClass)` I think

**Comment by gavinking** [comment]
This is now fixed.
